lib/flagutil/secure_url_test.go

```diff
@@ -0,0 +1,69 @@
+package flagutil
```

Comment:
In the tests I don't see the case when `authToken` will be secured. For example, `-snapshot.createURL="http://localhost:8428/snapshot/create?authKey=foobar"` - will it hide `authKey` value from the logs and /flags output?

Reply:
Yes. Basically, the change will mask the whole of the URL as `"secret"` in logs (Just like the remoteWriteURLs).
I have covered this case(authKey) in the latest commit.